src/components/Highlight/index.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import styled, { themed } from '../../utils/styled';
+import { StandardProps } from '../../common';
+
+export interface HighlightProps extends StandardProps {
+  /**
+   * String that needs to be rendered
+   */
+  text: string;
+  /**
+   * String that needs to be highlighted
+   */
+  highlight: string;
+  /**
+   * Defines if the search will be case insensitive
+   */
+  ignoreCase?: boolean;
+}
+
+const Highlighted = styled.span`
+  background-color: ${themed(props => props.theme.highlightColor)};
+`;
+
+/**
+ * Highlight component is meant to be a simple component to display text with highlighted search.
+ * Component will render a SPAN or series of SPAN with the content and highlights
```

Comment:
a ~~SPAN or~~ series of SPAN

Reply:
If there is no match it renders a single span

Reply:
really? as far as I can see the root is React.Fragment (which is not a span) and it contains a  series of spans.

Reply:
gotcha :)